src/Symfony/Component/HttpKernel/DataCollector/ExceptionDataCollector.php

```diff
@@ -22,16 +23,30 @@
  */
 class ExceptionDataCollector extends DataCollector
 {
+    private $flattener;
+
+    public function __construct(ExceptionFlattener $flattener = null)
+    {
+        $this->flattener = $flattener;
+    }
+
     /**
      * {@inheritdoc}
      */
     public function collect(Request $request, Response $response, \Exception $exception = null)
     {
-        if (null !== $exception) {
-            $this->data = array(
-                'exception' => FlattenException::create($exception),
-            );
+        if (null === $exception) {
+            return;
         }
+
+        $flattenException = $request->attributes->get('exception');
```

Comment:
when would it be set in the request ? the only case where I can find it in the PR is for the subrequest of the exception controller, where this attribute contains the exception which happened in the main request. but in this case, using it here is a bug, as it would be the wrong exception.